src/DataViewer.js

```diff
@@ -0,0 +1,96 @@
+import ExpandMoreIcon from '@material-ui/icons/ExpandMore';
+import ExpansionPanel from '@material-ui/core/ExpansionPanel';
+import ExpansionPanelDetails from '@material-ui/core/ExpansionPanelDetails';
+import ExpansionPanelSummary from '@material-ui/core/ExpansionPanelSummary';
+import React, { Component } from 'react';
+import ReactJson from 'react-json-view'
+import Typography from '@material-ui/core/Typography';
+import firebase from 'firebase';
+import Paper from 'material-ui/Paper';
+
+const DB_STATE = Object.freeze({
+  UNINITIALIZED: 0,
+  LOADING: 1,
+  LOADED: 2,
+});
+
+class DatabaseViewer extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      db_state: DB_STATE.UNINITIALIZED,
+      data: null,
+    };
+  }
+
+  handleExpanded(expanded) {
+    if (expanded) {
+      switch (this.state.db_state) {
+        case DB_STATE.UNINITIALIZED:
+          this.setState({ db_state: DB_STATE.LOADING });
+          firebase.database().ref("/").once("value", (snapshot) => {
+            // Check if user data firebase download is done yet.
+            var db_state = this.state.user_data === null
+                ? DB_STATE.LOADING : DB_STATE.LOADED;
+            this.setState({
+              db_state: db_state,
```

Comment:
Why wouldn't this **always** be `db_state: DB_STATE.LOADED`? The "check if user data firebase download is done yet" seems irrelevant -- if you enter this callback, then the firebase download must be done.